### PR TITLE
172031797 course overview

### PR DESCRIPTION
**Pull request by gracekfan**

### CODE REVIEW

File: app/controllers/professors_controller.rb [hunk 1]
```diff
@@ -12,8 +12,10 @@ def index
 
   # GET /professors/1
   # GET /professors/1.json
-  def avg_rev(course)
-    course.reviews.blank? ? 0 : course.reviews.average(:overall_rating).round(2)
+  def avg_rev(reviews)
+    sum = 0
+    reviews.each { |a| sum += a.overall_rating }
+    sum.to_f / reviews.length
   end
```
Comment (SaltyQuetzals) [inline app/controllers/professors_controller.rb:19]: This is identical code to `CoursesController#avg_rev`. Please refactor both functions into one shared function, that way modifying one modifies both.

File: app/controllers/professors_controller.rb [hunk 2]
```diff
@@ -44,7 +46,7 @@ def show
 
     if !@lowest_rated_course.blank? && !@lowest_rated_course.reviews.blank?
       @lowest_rated_course_review =
-        @lowest_rated_course.reviews.order('overall_rating DESC').first
+        @lowest_rated_course.reviews.order('overall_rating').first
```
Comment (SaltyQuetzals) [inline app/controllers/professors_controller.rb:49]: Either line 41 or this line isn't functioning as intended.

If you abstract these lines (removing mentions of highest/lowest) you get this:
```ruby
if !course.blank? && course.reviews.blank?
  course_review = course.reviews.order('overall_rating').first
end
```
If `order('overall_rating')` orders in ascending order, then `@highest_rated_course_review` is taking the lowest-rated course. If it orders in descending order, then `@lowest_rated_course_review` is taking the highest-rated course.

File: features/step_definitions/submit_review.rb [hunk 3]
```diff
@@ -3,7 +3,7 @@
 Given(/^I'm on the new review page$/) { visit new_review_path }
 
 And(/^I enter "([^"]*)" in the "Overall rating" field$/) do |_value|
-  page.choose(:review_overall_rating_1)
+  page.choose('review_overall_rating_' + _value)
```
Comment (SaltyQuetzals) [inline features/step_definitions/submit_review.rb:6]: Remove the underscore from `_value` in this, we are using the variable.

File: features/vote_review.feature [hunk 4]
```diff
@@ -68,7 +68,7 @@ Feature: Rate Reviews
     And I enter "5" in the "Professor other thoughts" field
     When I submit my Review form
     When I click the downvote button
-    Then I should see the number of downvotes equal to "1"
+    Then I should see the number of upvotes equal to "0"
```
Comment (SaltyQuetzals) [inline features/vote_review.feature:71]: Does a clean review start at 1 `upvote`? Or `0` upvotes? If it starts at `0`, isn't this checking default behavior?

File: features/vote_review.feature [hunk 5]
```diff
@@ -143,7 +143,6 @@ Feature: Rate Reviews
...
     When I click the downvote button
-    Then I should see the number of downvotes equal to "0"
-
+    Then I should see the number of upvotes equal to "0"
```
Comment (SaltyQuetzals) [inline features/vote_review.feature:148]: This doesn't seem to be undoing the downvote, the expectation is that the value stays the same despite clicking the `downvote` button.

File: features/step_definitions/submit_review.rb (hunk 3 above)
Comment (deshantar99) [reply features/step_definitions/submit_review.rb]: Fixed

**Pull request merged**